src/auto/uri.rs

```diff
@@ -0,0 +1,530 @@
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_userinfo(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_userinfo(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn parse_relative(&self, uri_ref: &str, flags: UriFlags) -> Result<Uri, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = glib_sys::g_uri_parse_relative(
+                self.to_glib_none().0,
+                uri_ref.to_glib_none().0,
+                flags.to_glib(),
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(from_glib_full(ret))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    fn to_string(&self) -> GString {
+        unsafe { from_glib_full(glib_sys::g_uri_to_string(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn to_string_partial(&self, flags: UriHideFlags) -> Option<GString> {
+        unsafe {
+            from_glib_full(glib_sys::g_uri_to_string_partial(
+                self.to_glib_none().0,
+                flags.to_glib(),
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn build(
+        flags: UriFlags,
+        scheme: &str,
+        userinfo: Option<&str>,
+        host: Option<&str>,
+        port: i32,
+        path: &str,
+        query: Option<&str>,
+        fragment: Option<&str>,
+    ) -> Option<Uri> {
+        unsafe {
+            from_glib_full(glib_sys::g_uri_build(
+                flags.to_glib(),
+                scheme.to_glib_none().0,
+                userinfo.to_glib_none().0,
+                host.to_glib_none().0,
+                port,
+                path.to_glib_none().0,
+                query.to_glib_none().0,
+                fragment.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn build_with_user(
+        flags: UriFlags,
+        scheme: &str,
+        user: Option<&str>,
+        password: Option<&str>,
+        auth_params: Option<&str>,
+        host: Option<&str>,
+        port: i32,
+        path: &str,
+        query: Option<&str>,
+        fragment: Option<&str>,
+    ) -> Option<Uri> {
+        unsafe {
+            from_glib_full(glib_sys::g_uri_build_with_user(
+                flags.to_glib(),
+                scheme.to_glib_none().0,
+                user.to_glib_none().0,
+                password.to_glib_none().0,
+                auth_params.to_glib_none().0,
+                host.to_glib_none().0,
+                port,
+                path.to_glib_none().0,
+                query.to_glib_none().0,
+                fragment.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn error_quark() -> Quark {
```

Comment:
This is would go away once the `UriError` enum has to correct error domain attached to it

Reply:
It doesn't seems to be or did I do something wrong?

Reply:
Maybe just put it into the ignore list for now then :) Can be fixed properly while gir is fixed

Reply:
Putting it in the ignore list doesn't seem to do anything, should I mark the enum as manual and fix the code for now?

Reply:
I guess just edit the manual code here now so we can finally get this in, and I'll work on fixing gir this weekend or tomorrow so we don't have this problem anymore in the future. Easy enough to fix there, just need to find 20 minutes of free time.

@GuillaumeGomez What do you think?

Reply:
Manual until fixed (if that was the question :) ). Or fix gir then regen this one (I tend to prefer doing it this way).

Reply:
Then let's wait until the weekend unless @bilelmoussaoui wants to fix gir

Reply:
I can look at fixing gir today if I have +/- an idea of how the enum configuration would be? At least for this specific use case

Reply:
See my comments in the issue. It's not about configuration, it's about using that field in the XML file correctly

Reply:
Actually, my bad. I was trying to ignore this function in the UriError enum instead of Uri itself. I've ignored the function for now until I (or someone else) get the time to look at gir issue